department-of-veterans-affairs/vets-website
# MHV-53237 MHV-54082 Fixed redirection test for unauthenticated users

## Summary

- Fixed a broken unit test for redirecting unauthenticated users
- Mocked API calls in the App.jsx unit tests

## Related issue(s)

### [MHV-53237](https://jira.devops.va.gov/browse/MHV-53237) - Fix no-auth frontend unit test for App.jsx ###

> Calling a state change function ("setPaths(navPaths);" from within a useEffect caused one of the unit tests to fail:
> 
> App.unit.spec.jsx - "user is not logged in"
> 
> Possibly due to some asynchronicity. I was unable to get the test working with the window.location check. Perhaps we need a better option to test the behavior when a user is not logged in.
> 
> User Story:  As a SM user, I want to    so that I can
> 
> 
> 
> 
> GIVEN:
> 
> WHEN:
> 
> THEN:
> 
> 
> 
> 
> Feature Flag Y/N ? N  N
> 
> Manual Testing Y/N ? N N
> 
> Automated Testing Y/N ? Y-Timothy
> 
> Accessibility Testing Y/N ? N
> 
> UCD Validation Y/N N
> 
> 
> 
> 
> Acceptance Criteria:
> 
> AC1  Implement a better option to test the behavior when a user is not logged in
> 
> AC2 
> 
> AC3
> AC4
> AC5
> 
> 
> 
> 
> Links to UCD:
> 
> Sketch Link:
> 
> User Flow:
> 
> Mobile:
> 
> Desktop:
> 
> Other Design Notes:
> 
> 
> 
> 
> Architectural Notes: Always check for the latest designs in Figma

### [MHV-54082](https://jira.devops.va.gov/browse/MHV-54082) - Mock the call to the status endpoint when we are testing <App> ###

> Mock the call to the status endpoint when we are testing <App>
> 
> Currently, unit tests that render <App> are taking a long time to complete because they are attempting to hit the status endpoint. Instead, we should mock that endpoint.
> https://dsva.slack.com/archives/C04ER7MHX5E/p1706541698875129?thread_ts=1705958091.259729&cid=C04ER7MHX5E
> 
> 
> 
> 
> 
> 
> 
> Feature Flag Y/N ? N
> 
> Manual Testing Y/N ? N
> 
> Automated Testing Y/N ? N
> 
> Accessibility Testing Y/N ? N
> 
> UCD Validation Y/N N
> 
> 
> 
> 
> Acceptance Criteria:
> 
> AC1  Endpoint is mocked
> 
> AC2
> 
> AC3
> AC4
> AC5
> 
> 
> 
> 
> Links to UCD:
> 
> Sketch Link:
> 
> User Flow:
> 
> Mobile:
> 
> Desktop:
> 
> Other Design Notes:
> 
> 
> 
> 
> Architectural Notes: Always check for the latest designs in Figma

## Testing done

- This PR is fully testing related.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

MHV Medical Records

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication
